src/test/java/it/reply/orchestrator/service/CloudProviderEndpointServiceTest.java

```diff
@@ -0,0 +1,136 @@
+package it.reply.orchestrator.service;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+import it.reply.orchestrator.controller.ControllerTestUtils;
+import it.reply.orchestrator.dal.entity.Deployment;
+import it.reply.orchestrator.dto.CloudProvider;
+import it.reply.orchestrator.dto.CloudProviderEndpoint;
+import it.reply.orchestrator.dto.RankCloudProvidersMessage;
+import it.reply.orchestrator.dto.CloudProviderEndpoint.IaaSType;
+import it.reply.orchestrator.dto.cmdb.CloudService;
+import it.reply.orchestrator.dto.cmdb.CloudServiceData;
+import it.reply.orchestrator.dto.cmdb.CmdbDataWrapper;
+import it.reply.orchestrator.dto.cmdb.Type;
+import it.reply.orchestrator.dto.deployment.PlacementPolicy;
+import it.reply.orchestrator.dto.ranker.RankedCloudProvider;
+import it.reply.orchestrator.exception.service.DeploymentException;
+import it.reply.utils.json.JsonUtility;
+
+public class CloudProviderEndpointServiceTest {
+	
+	@InjectMocks
+	CloudProviderEndpointServiceImpl cloudProviderEndpointServiceImpl = new CloudProviderEndpointServiceImpl();
+
+	@Test
+	public void chooseCloudProvider() throws Exception {
+		Deployment deployment = ControllerTestUtils.createDeployment();
+		RankCloudProvidersMessage rcpm = new RankCloudProvidersMessage();
+		
+		List<RankedCloudProvider> listRankedCloudProvider = new ArrayList<>();
+		
+		RankedCloudProvider rankedCloudProvider1 = new RankedCloudProvider();
+		rankedCloudProvider1.setRank(100);
+		RankedCloudProvider rankedCloudProvider2 = new RankedCloudProvider();
+		rankedCloudProvider2.setRank(400);
+		RankedCloudProvider rankedCloudProvider3 = new RankedCloudProvider();
+		rankedCloudProvider3.setRank(200);
+		
+		listRankedCloudProvider.add(rankedCloudProvider1);
+		listRankedCloudProvider.add(rankedCloudProvider2);
+		listRankedCloudProvider.add(rankedCloudProvider3);
+
+		rcpm.setRankedCloudProviders(listRankedCloudProvider);
+				
+		assertEquals(cloudProviderEndpointServiceImpl.chooseCloudProvider(deployment, rcpm), rankedCloudProvider2);
+	}
+	
+	@Test(expected = DeploymentException.class)
+	public void failChooseCloudProvider() throws Exception {
+		Deployment deployment = ControllerTestUtils.createDeployment();
+		RankCloudProvidersMessage rcpm = new RankCloudProvidersMessage();
+
+		
+		assertEquals(cloudProviderEndpointServiceImpl.chooseCloudProvider(deployment, rcpm), null);

```

Comment:
the assertEquals() never get called, as the Exception get thown before it. Can be removed